xds/internal/resolver/xds_resolver.go

```diff
@@ -80,7 +80,7 @@ func (b *xdsResolverBuilder) Build(t resolver.Target, cc resolver.ClientConn, rb
 	r.client = client
 
 	// Register a watch on the xdsClient for the user's dial target.
-	cancelWatch := r.client.WatchService(r.target.Endpoint, r.handleServiceUpdate)
+	cancelWatch := watchService(r.client, r.target.Endpoint, r.handleServiceUpdate, r.logger)
```

Comment:
Would it be better to make `watchService` a method on the `xdsResolver`? That way we wont have to pass all these arguments.

Reply:
I did that at first. But it made testing hard. Because you will need to make a `xdsResolver`, and replace it's fields to the test xds client, test update handler. It end up easier to test a function like this.